### DISCUSSION TITLE
Support being connected to wrong chain on writes

### DISCUSSION THREAD

**Discussion by roninjin10** [wevm/viem#224] [Idea / Feature Request] [answered]
Throwing error should be default.   But would be good to be able to pass in an option here to cause one of these behaviors:

**Auto switch chains**
If on wrong chain we automatically request to switch chains

**Auto switch chains only sometimes**
Some wallets support [eip-1637](https://github.com/ethereum/EIPs/issues/1637).   Attempt to send tx if we know we are connected to one of those wallets.   Otherwise fall back to auto requesting a chain switch

**Comment by jxom** [comment, marked as answer]
I'd like to avoid side-effects in `sendTransaction` as it doesn't play nicely with WalletConnect & iOS. 

Folks can do this themselves via `switchChain` if they really want to. :)

**Reply by roninjin10** [reply]
If you didn't switch chains but still handled eip-1637 case that isn't a side effect

**Reply by jxom** [reply]
I don't believe any wallets support 1637 right now.

**Reply by roninjin10** [reply]
ah wierd I thought rainbow wallet did for some reason

**Reply by jxom** [reply]
Maybe at some point we can support [CAIP-27](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-27.md) (the modern day 1637), but most wallets do not support that either.

**Reply by roninjin10** [reply]
Oh wow this session token part I've never seen before.   This proposal would be amazing!

**Reply by roninjin10** [reply]
Wait "most" wallets?   If any wallets support it I think Viem should support it too to try to get more wallets to support it faster
